### PR TITLE
chore: Replaced apt with apt-get

### LINKED ISSUE #787: [REQ] Use apt-get instead of apt
### What kind of request is this?

None

### What is your request or suggestion?

Investigate using apt-get for more stability instead of apt for Debian patching.

### Are you willing to submit PRs to contribute to this feature request?

- [ ] Yes, I am willing to implement it.

### PR DESCRIPTION
**Pull request by prakrit55**
Describe the changes in this pull request using active verbs such as _Add_, _Remove_, _Replace_ ...
* Replaced apt with apt-get

Closes #787

### CODE REVIEW

**Comment by prakrit55** [comment]
@ashnamehrotra, can you check it once, if it is okay

File: pkg/pkgmgr/dpkg.go [hunk 1]
```diff
@@ -319,29 +319,29 @@ func (dm *dpkgManager) installUpdates(ctx context.Context, updates unversioned.U
...
 		pkgStrings := []string{}
 		for _, u := range updates {
 			pkgStrings = append(pkgStrings, u.Name)
 		}
-		installCmd = fmt.Sprintf(aptInstallTemplate, strings.Join(pkgStrings, " "))
+		installCmd = fmt.Sprintf(aptGetInstallTemplate, strings.Join(pkgStrings, " "))
 	} else {
 		// if updates is not specified, update all packages
-		installCmd = `sh -c "output=$(apt upgrade -y && apt clean -y && apt autoremove 2>&1); if [ $? -ne 0 ]; then echo "$output" >>error_log.txt; fi"`
+		installCmd = `sh -c "output=$(apt-get upgrade -y && apt-get clean -y && apt-get autoremove 2>&1); if [ $? -ne 0 ]; then echo "$output" >>error_log.txt; fi"`
```
Comment (ashnamehrotra) [inline pkg/pkgmgr/dpkg.go:330]: it seems like this command is not downloading the packages, resulting in the error for the test cases with no report

Comment (prakrit55) [reply pkg/pkgmgr/dpkg.go:330]: let me check

Comment (prakrit55) [reply pkg/pkgmgr/dpkg.go:330]: @ashnamehrotra, could you check [it](https://github.com/project-copacetic/copacetic/actions/runs/11266328802/job/31329688791?pr=804#step:9:107) once, and [this](https://github.com/project-copacetic/copacetic/actions/runs/11266328802/job/31329688791?pr=804#step:9:263) too

Comment (ashnamehrotra) [reply pkg/pkgmgr/dpkg.go:330]: it looks like results manifest file is not found, so the packages are not found/downloaded correctly. There is also a log output `line 2: packages.txt: No such file or directory`. Would you be able to run the apt-get commands manually in a container to see if they are working as expected?

File: pkg/pkgmgr/dpkg.go [hunk 2]
```diff
@@ -411,7 +411,7 @@ func (dm *dpkgManager) unpackAndMergeUpdates(ctx context.Context, updates unvers
 							while IFS=':' read -r package version; do
 								pkg_name=$(echo "$package" | sed 's/^"\(.*\)"$/\1/')
 								pkg_version=$(echo "$version" | sed 's/^"\(.*\)"$/\1/')
-								latest_version=$(apt show $pkg_name 2>/dev/null | awk -F ': ' '/Version:/{print $2}')
+								latest_version=$(apt-get show $pkg_name 2>/dev/null | awk -F ': ' '/Version:/{print $2}')
```
Comment (sozercan) [inline pkg/pkgmgr/dpkg.go:414]: `apt-get show` is not a valid command

File: website/docs/faq.md [hunk 3]
```diff
@@ -43,7 +43,7 @@ Experimental: This feature might change without preserving backwards compatibili
 
 :::
 
-Copa does not support replacing the repositories in the package managers with alternatives. Images must already use the intended package repositories. For example, for debian, updating `/etc/apt/sources.list` from `http://archive.ubuntu.com/ubuntu/` to a mirror, such as `https://mirrors.wikimedia.org/ubuntu/`.
+Copa does not support replacing the repositories in the package managers with alternatives. Images must already use the intended package repositories. For example, for debian, updating `/etc/apt-get/sources.list` from `http://archive.ubuntu.com/ubuntu/` to a mirror, such as `https://mirrors.wikimedia.org/ubuntu/`.
```
Comment (sozercan) [inline website/docs/faq.md:46]: there's no /var/apt-get repo

File: pkg/pkgmgr/dpkg.go [hunk 4]
```diff
@@ -24,7 +24,7 @@ const (
 	dpkgLibPath      = "/var/lib/dpkg"
 	dpkgStatusPath   = dpkgLibPath + "/status"
 	dpkgStatusFolder = dpkgLibPath + "/status.d"
-	dpkgDownloadPath = "/var/cache/apt/archives"
+	dpkgDownloadPath = "/var/cache/apt-get/archives"
```
Comment (sozercan) [inline pkg/pkgmgr/dpkg.go:27]: no `/var/cache/apt-get`

please make sure these exist before find and replace

**Review by sozercan — CHANGES_REQUESTED** [review]
see comments

**Comment by sozercan** [comment]
@prakrit55 looks like there's a lint issue

**Comment by prakrit55** [comment]
> @prakrit55 looks like there's a lint issue

there are some other instances of apt and to be rechecked, let me cover those too

**Review by sozercan — APPROVED** [review]
Thank you! LGTM

**Pull request merged**